crates/interpreter/src/valid.rs

```diff
@@ -856,8 +883,12 @@ impl<'a, 'm> Expr<'a, 'm> {
         branch
```

Comment:
We probably need to add `self.stack().len()` to `branch.stack`, in addition of setting `val_cnt`.

Reply:
`prev_stack` has been updated so in `push_label`. Why do we need to do this again to `branch.stack`?

Reply:
You suggested adding `branch.stack += self.stack().len();` in `branch_source()`.

`branch_source()` calls `branch_target()` which set `SideTableBranch::stack` to be `self.immutable_label().prev_stack`. In `push_label()`, `prev_stack` is `prev_label.prev_stack + prev_label.stack.len()`, which seems the same as your suggestion here. Why do we need to do it again?

Reply:
I see. It's not the same stack because it's not the same label.

In `push_label()` we record `prev_stack` as the stack size up to (but excluding) the label we're pushing. This is equivalent to the stack size up to (but excluding) the current label (before we push) plus the stack size of that current label, hence: `prev_stack = prev_label.prev_stack + prev_label.stack.len()`.

In `branch_source()` we record `stack` as the stack size at the current program point. This is equivalent to the stack size up to (but excluding) the current label (which we get from `branch_target()`) plus the stack size of the current label, hence: `branch.stack += self.stack().len()`.